### PR TITLE
Filter programs based on sector, subsector, marker, submarker

### PR DESCRIPTION
**Pull request by samshara**

### CODE REVIEW

File: src/components/ProgramSelector/index.tsx [hunk 1]
```diff
@@ -195,13 +195,75 @@ function ProgramSelector(props: Props) {
...
+            const sectorKeys = sectorOptions
+                ?.filter(sector => selectedSector?.includes(sector.key))
+                .map(sector => sector.id);
+
+            const subSectorKeys = subSectorOptions
+                ?.filter(subSector => selectedSector?.includes(subSector.key))
+                .map(subSector => subSector.id);
+
+            const filteredProgramsBySector = programs.filter(program => (
+                program.sector.some(sector => sectorKeys?.includes(sector))
```
Comment (tnagorra) [inline src/components/ProgramSelector/index.tsx]: We should skip filtering if sectorKeys is empty

File: src/components/ProgramSelector/index.tsx [hunk 2]
```diff
@@ -195,13 +195,75 @@ function ProgramSelector(props: Props) {
...
+            ));
+
+            const filteredProgramsBySubSector = programs.filter(program => (
+                program.subSector.some(subSector => subSectorKeys?.includes(subSector))
```
Comment (tnagorra) [inline src/components/ProgramSelector/index.tsx]: We should skip filtering if subSectorKeys is empty

File: src/components/ProgramSelector/index.tsx [hunk 3]
```diff
@@ -195,13 +195,75 @@ function ProgramSelector(props: Props) {
...
+            const markerKeys = markerOptions
+                ?.filter(marker => selectedMarker?.includes(marker.key))
+                .map(marker => marker.id);
+
+            const subMarkerKeys = subMarkerOptions
+                ?.filter(subMarker => selectedMarker?.includes(subMarker.key))
+                .map(subMarker => subMarker.id);
+
+            const filteredProgramsByMarker = programs.filter(program => (
+                program.markerCategory.some(marker => markerKeys?.includes(marker))
```
Comment (tnagorra) [inline src/components/ProgramSelector/index.tsx]: We should skip filtering if markerKeys is empty

File: src/components/ProgramSelector/index.tsx [hunk 4]
```diff
@@ -195,13 +195,75 @@ function ProgramSelector(props: Props) {
...
+            ));
+
+            const filteredProgramsBySubMarker = programs.filter(program => (
+                program.markerValue.some(subMarker => subMarkerKeys?.includes(subMarker))
```
Comment (tnagorra) [inline src/components/ProgramSelector/index.tsx]: We should skip filtering if subMarkerKeys is empty

**Review by tnagorra — CHANGES_REQUESTED** [review]
Need to fix filtering logic

**Review by samshara — COMMENTED** [review]
fixed filtering logic

**Pull request closed without merging**